portability-core/src/test/java/org/dataportabilityproject/serviceProviders/flickr/FlickrPhotoServiceTest.java

```diff
@@ -93,6 +106,40 @@ public void toCommonPhoto() {
     assertThat(photoModel.getMediaType()).isEqualTo("image/jpeg");
   }
 
+  @Test
+  public void importStoresAlbumsInJobCache() throws IOException, FlickrException {
+    // Set up input: a single photo album with a single photo
+    PhotosModelWrapper wrapper = new PhotosModelWrapper(Arrays.asList(PHOTO_ALBUM),
+        Arrays.asList(PHOTO_MODEL), new ContinuationInformation(null, null));
+
+    // Set up mocks
+    when(imageStreamProvider.get(FETCHABLE_URL)).thenReturn(inputStream);
+
+    when(uploader
+        .upload(Matchers.any(BufferedInputStream.class), Matchers.any(UploadMetaData.class)))
+        .thenReturn(FLICKR_PHOTO_ID);
+
+    String flickrAlbumTitle = FlickrPhotoService.FLICKR_ALBUM_PREFIX + ALBUM_NAME;
+    Photoset photoSet = initializePhotoset(FLICKR_ALBUM_ID, flickrAlbumTitle, ALBUM_DESCRIPTION);
+    when(photosetsInterface.create(flickrAlbumTitle, ALBUM_DESCRIPTION, FLICKR_PHOTO_ID))
+        .thenReturn(photoSet);
+
+    // Run test
+    photoService.importItem(wrapper);
+
+    // Verify the mocks were called correctly
```

Comment:
nit: might rephrase this as what we specifically want to test was called, e.g.

"Verify the photo was uploaded" etc